sdk/core/System.Net.ClientModel/api/System.Net.ClientModel.net6.0.cs

```diff
@@ -0,0 +1,407 @@
+        public abstract System.Threading.Tasks.ValueTask ProcessAsync(System.Net.ClientModel.Core.ClientMessage message, System.Net.ClientModel.Core.PipelineEnumerator pipeline);
+    }
+    public abstract partial class PipelineTransport : System.Net.ClientModel.Core.PipelinePolicy
+    {
+        protected PipelineTransport() { }
+        public abstract System.Net.ClientModel.Core.ClientMessage CreateMessage();
+        public abstract void Process(System.Net.ClientModel.Core.ClientMessage message);
+        public override void Process(System.Net.ClientModel.Core.ClientMessage message, System.Net.ClientModel.Core.PipelineEnumerator pipeline) { }
+        public abstract System.Threading.Tasks.ValueTask ProcessAsync(System.Net.ClientModel.Core.ClientMessage message);
+        public override System.Threading.Tasks.ValueTask ProcessAsync(System.Net.ClientModel.Core.ClientMessage message, System.Net.ClientModel.Core.PipelineEnumerator pipeline) { throw null; }
+    }
+    public abstract partial class RequestBodyContent : System.IDisposable
+    {
+        protected RequestBodyContent() { }
+        public static System.Net.ClientModel.Core.RequestBodyContent Create(System.BinaryData value) { throw null; }
+        public static System.Net.ClientModel.Core.RequestBodyContent Create(System.Net.ClientModel.Core.IModel<object> model, System.Net.ClientModel.ModelReaderWriterOptions? options = null) { throw null; }
+        public abstract void Dispose();
+        public abstract bool TryComputeLength(out long length);
+        public abstract void WriteTo(System.IO.Stream stream, System.Threading.CancellationToken cancellationToken);
+        public abstract System.Threading.Tasks.Task WriteToAsync(System.IO.Stream stream, System.Threading.CancellationToken cancellationToken);
+    }
+    public partial class ResponseBufferingPolicy : System.Net.ClientModel.Core.PipelinePolicy
+    {
+        public ResponseBufferingPolicy(System.TimeSpan networkTimeout) { }
+        public override void Process(System.Net.ClientModel.Core.ClientMessage message, System.Net.ClientModel.Core.PipelineEnumerator pipeline) { }
+        public override System.Threading.Tasks.ValueTask ProcessAsync(System.Net.ClientModel.Core.ClientMessage message, System.Net.ClientModel.Core.PipelineEnumerator pipeline) { throw null; }
+        public static void SetBufferResponse(System.Net.ClientModel.Core.ClientMessage message, bool bufferResponse) { }
+        public static void SetNetworkTimeout(System.Net.ClientModel.Core.ClientMessage message, System.TimeSpan networkTimeout) { }
+        public static bool TryGetBufferResponse(System.Net.ClientModel.Core.ClientMessage message, out bool bufferResponse) { throw null; }
+        public static bool TryGetNetworkTimeout(System.Net.ClientModel.Core.ClientMessage message, out System.TimeSpan networkTimeout) { throw null; }
```

Comment:
Could these be extension methods on ClientMessage? I imagine the list of APIs here could grow to support any scenario where we want a public API for setting internal message properties.

Reply:
I like the question - I kind of like localizing the properties we put in the message property bag to the relevant policies, but I have a bunch of concerns around the number of different ways we have to ultimately set options so if we could get the number of "ways" down to just a few and be principled about what types of options fall into which categories, I'd be happier with a simplified story around this.